druid/src/core.rs

```diff
@@ -1013,10 +964,30 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
                         _ => false,
                     }
                 }
-                InternalLifeCycle::ParentWindowOrigin => {
-                    self.state.parent_window_origin = ctx.widget_state.window_origin();
-                    self.state.needs_window_origin = false;
-                    true
+                InternalLifeCycle::RouteViewContextChanged(view_context) => {
+                    if self.state.view_context_changed {
+                        self.lifecycle(
+                            ctx,
+                            &LifeCycle::ViewContextChanged(*view_context),
+                            data,
+                            env,
+                        );
+                        self.state.view_context_changed = false;
+                        self.state.children_view_context_changed = false;
```

Comment:
1) You are right, they are redundant.

2) Lets say we have a widget A which has a child widget B. 
When A receives a `Lifecycle::ViewContextChnaged` event, it can first call B.set_origin and then propagate the event.
This works and there are widgets were this is needed.
But doing it the other way round results in inconsistent state.
In the first case the the state is correctly handled and we don't need to set the flags to `true`.
And in the second case setting them to true doesn't help either, since the widget isn't getting another `ViewContextChanged` event, before the next event.

I probably should add this to the docs of `set_origin`.